src/js/tasks-function.js

```diff
@@ -0,0 +1,223 @@
+      this.displayTask();
+    }
+  }
+
+  updateTask(item, description) {
+    if (this.taskData[item]) {
+      this.taskData[item].description = description;
+    }
+    this.taskData.forEach((elem, index) => {
+      elem.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      this.displayTask();
+    }
+  }
+
+  clearAllCompletTask = () => {
+    this.taskData = this.taskData.filter((elem) => elem.completed === false);
+    this.taskData.forEach((e, index) => {
+      e.index = index;
+    });
+    if (typeof window !== 'undefined') {
+      localStorage.setItem('TodoListDB', JSON.stringify(this.taskData));
+      // eslint-disable-next-line no-unused-expressions
+      window.location.reload()();
+    }
+  }
+}
+
+// Add New Task and Display it
+let index = 0;
+const completed = false;
+const entryTask = new Todolist();
+
+const newItemInput = document.querySelector('#new-item');
+// const taskList = document.querySelector('.task-list');
+const btnAddNewTask = document.querySelector('.btn-add');
+btnAddNewTask.addEventListener('click', () => {
+  if (newItemInput.value === '') {
+    btnAddNewTask.setCustomValidity('This is required field!');
+  } else {
+    entryTask.addtask(newItemInput.value, completed, index);
+    entryTask.displayTask();
+    newItemInput.value = '';
+    index += 1;
+  }
+});
+
+window.onload = () => {
+  entryTask.taskData = JSON.parse(localStorage.getItem('TodoListDB' || []));
+  if (entryTask.taskData === null) {
+    entryTask.taskData = [];
+    return;
+  }
+  entryTask.displayTask();
+};
+
+const clearAllTaskBtn = document.querySelector('.clear-all');
+clearAllTaskBtn.addEventListener('click', (e) => {
+  entryTask.clearAllCompletTask();
+  e.preventDefault();
+  window.location.reload();
+});
+// const createItem = (item) => {
+//   tasksArr.push(item.value);
+//   localStorage.setItem('tasks', JSON.stringify(tasksArr));
+//   location.reload();
+// };
+// document.querySelector('.btn-add').addEventListener('click', Todolist.addtask(new Tasks()));
+
+// const displayItem = () => {
+//   let items = '';
+//   for (let i = 0; i < tasksArr.length; i+ =1) {
+//     const item = tasksArr[i];
+//     items += `<li class="task flex">
+//     <input type="checkbox" id="checked" name="checked" >
+//     <label class="label" id="label" for="checked">${item}</label>
+//     <i class="fa-solid fa-ellipsis-vertical ellipsis"></i>
+//     <i class="fa-solid fa-ellipsis-vertical deleteBtn"></i>
+//     </li>`;
+//     taskList.innerHTML = items;
+//   }
+// };
+
+// window.onload = displayItem();
+// activateDeleteListeners();
+
+// let activateDeleteListeners = () => {
+//   const deleteBtn = document.querySelectorAll('.deleteBtn');
+//   deleteBtn.forEach((db, i) => {
+
+//   });
+// };
+// console.log(tasksArr);
+// window.onload = Todolist.displayTask
+// console.log('ggggggggggggg');
```

Comment:
- [ ] Kindly delete any commented code as that makes your project look unprofessional.